src/Agent/NewRelic/Profiler/Configuration/Configuration.h

```diff
@@ -47,24 +47,24 @@ namespace NewRelic { namespace Profiler { namespace Configuration {
             , _ignoreList(new IgnoreInstrumentationList())
         {
             try {
-                rapidxml::xml_document<xchar_t> globalNewRelicConfigurationDocument;
-                globalNewRelicConfigurationDocument.parse<rapidxml::parse_trim_whitespace | rapidxml::parse_normalize_whitespace>(const_cast<xchar_t*>(globalNewRelicConfiguration.c_str()));
+                auto globalNewRelicConfigurationDocument = std::make_shared<rapidxml::xml_document<xchar_t>>();
+                globalNewRelicConfigurationDocument->parse<rapidxml::parse_trim_whitespace | rapidxml::parse_normalize_whitespace>(const_cast<xchar_t*>(globalNewRelicConfiguration.c_str()));
 
-                auto globalNewRelicConfigurationNode  = GetConfigurationNode(globalNewRelicConfigurationDocument);
+                auto globalNewRelicConfigurationNode = GetConfigurationNode(globalNewRelicConfigurationDocument);
                 if (globalNewRelicConfigurationNode == nullptr) 
                 {
                     LogError(L"Unable to locate configuration node in the global newrelic.config file.");
                     throw ConfigurationException();
                 }
 
                 auto appliedNewRelicConfigurationNode = globalNewRelicConfigurationNode;
+                auto localNewRelicConfigurationDocument = std::make_shared<rapidxml::xml_document<xchar_t>>();
 
                 if (localNewRelicConfiguration.second)
                 {
                     try
                     {
-                        rapidxml::xml_document<xchar_t> localNewRelicConfigurationDocument;
```

Comment:
This may have been an actual bug. We were using elements from `localNewRelicConfigurationDocument` after it went out of scope. When I moved it to the heap it started to crash, which is why I moved its allocation.